### ISSUE TITLE
Apparent false positive: Value is immediately overriden

### ISSUE THREAD

**Issue by shmeeps** [kalessil/phpinspectionsea#1642]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 4.0.6 |
| Language level | e.g. PHP 7.1                        |

Current behaviour
--
New array element assignment within an if-block can flag a subsequent assignment of a variable as "immediately overridden" when the variable is used in between each assignment. This appears to only occur when there is a function call between the first assignment and use. Removing said function call or adding an else-block to the statement will remove the warning. Potentially related to #1588. Distilled example:

```
<?php

$items = array();

if (true) {
    $item = 'test';    // No warning
    var_dump(true);    // Any function / $obj->func call causes the warning. Removing this line removes the warning.
    $items[] = $item;
}                      // Attaching an else block also removes the warning

$item = 'test2';       // Immediately overridden warning
$items[] = $item;
```

Expected behaviour
--
Line 11 should not be flagged as "immediately overriden" as the value assigned on line 6 is used on line 8, even though there is an unrelated function call between them.

Environment details
---

![image](https://user-images.githubusercontent.com/1050677/95392812-0c1dfe00-08bf-11eb-8fd1-156de9211569.png)
